roles/consul/tasks/uninstall.yml

```diff
@@ -0,0 +1,40 @@
+---
+
+- name: Uninstall Consul monitoring resources
+  command: >-
+    kubectl delete {{ item.kind }} {{ item.name }} --namespace {{ consul_release_namespace }}
+  register: consul_delete_monitoring_resource
+  changed_when: consul_delete_monitoring_resource.rc == 0
+  failed_when: >-
+    consul_delete_monitoring_resource.rc != 0 and
+    "NotFound" not in consul_delete_monitoring_resource.stderr
+  loop:
+    - kind: prometheusrule
+      name: consul
+    - kind: configmap
+      name: consul-grafana-dashboard
+    - kind: podmonitor
+      name: "{{ consul_release_name }}-client"
+    - kind: servicemonitor
+      name: "{{ consul_release_name }}-server"
+  loop_control:
+    label: "{{ item.kind }}/{{ item.name }}"
+
+- name: Remove Consul ingress
+  command: >-
+    kubectl delete ingress
+      {{ admin_dashboard_ingress_consul_name | default('consul') }}
+      --namespace {{ admin_dashboard_ingress_consul_namespace | default('azimuth') }}
+  register: consul_delete_ingress
+  changed_when: consul_delete_ingress.rc == 0
+  failed_when: >-
+    consul_delete_ingress.rc != 0 and
+    "NotFound" not in consul_delete_ingress.stderr
+
+- name: Uninstall Consul from target Kubernetes cluster
+  kubernetes.core.helm:
+    release_namespace: "{{ consul_release_namespace }}"
+    release_name: "{{ consul_release_name }}"
+    state: absent
+    wait: yes
+    wait_timeout: "{{ consul_wait_timeout }}"
```

Comment:
Do we need another task after this to delete the Consul server stateful set PVCs?

Reply:
I was going to leave them for now, just in case